src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -54,43 +57,56 @@ class MultitableFlowsSpec extends HealthCheckSpecification {
     @Autowired
     Provider<TraffExamService> traffExamProvider
 
+    final String FLOW_DOESNT_PASS_TRAFFIC_ASSERTION_MESSAGE =
+            "Flow did not pass traffic at least in one direction when expected to pass"
+    final String FLOW_ISNT_PINGABLE_ASSERTION_MESSAGE = "Flow was not pingable when expected to be"
+
     @Tidy
     @Tags([LOW_PRIORITY])
-    def "System can use both single-table and multi-table switches in flow path at the same time, change switch table \
-mode with existing flows and hold flows of different table-mode types"() {
+    def "System can use both single-table and multi-table switches in flow path at the same time"() {
         given: "A potential flow on a path of 4 switches: multi -> single -> multi -> single"
-        List<PathNode> desiredPath = null
-        List<Switch> involvedSwitches = null
-        def allTraffgenSwitchIds = topology.activeTraffGens*.switchConnected.dpId ?:
-                assumeTrue(false, "Should be at least two active traffgens connected to switches")
-        def swPair = topologyHelper.allNotNeighboringSwitchPairs.collectMany { [it, it.reversed] }.find { pair ->
-            desiredPath = pair.paths.find { path ->
-                involvedSwitches = pathHelper.getInvolvedSwitches(path)
-                //4 switches total. First and third switches should allow multi-table
-                involvedSwitches.size() == 4 && involvedSwitches[0].dpId in allTraffgenSwitchIds &&
-                        involvedSwitches[-1].dpId in allTraffgenSwitchIds &&
-                        involvedSwitches.every { it.features.contains(SwitchFeature.MULTI_TABLE) }
+        def (involvedSwitches, swPair, desiredPath, initSwProps) =
+                                                "get a potential flow on a path of 4 switches: multi and single"()
+        when: "Create the prepared hybrid flow"
+        def flow = flowHelperV2.randomFlow(swPair)
+        flowHelperV2.addFlow(flow)
+        assert PathHelper.convert(northbound.getFlowPath(flow.flowId)) == desiredPath
+
+        then: "Created flow is valid"
+        northbound.validateFlow(flow.flowId).each { assert it.asExpected }
+
+        and: "Involved switches pass switch validation"
+        "validate switches"(involvedSwitches)
+        assert "is flow pingable?"(flow), FLOW_ISNT_PINGABLE_ASSERTION_MESSAGE
```

Comment:
This line is not about `Involved switches pass switch validation`. I think you should move it to section `and: "Flow is pingable"` (like in next test)